A bunch of bug fixes. (#4594)
pulumi/docs: content/docs/intro/cloud-providers/kubernetes/_index.md

- .NET: [`Pulumi.Kubernetes`](https://www.nuget.org/packages/Pulumi.Kubernetes)

The Kubernetes provider is open source and available in the [pulumi/pulumi-kubernetes](https://github.com/pulumi/pulumi-kubernetes) repo.

### Configuration

The Kubernetes provider accepts the following configuration settings.  These can be provided to the default Kubernetes provider via `pulumi config set kubernetes:<option>`, or passed to the constructor of `new kubernetes.Provider` to construct a specific instance of the Kubernetes provider.

- `cluster`: (Optional) If present, the name of the kubeconfig cluster to use.
- `context`: (Optional) If present, the name of the kubeconfig context to use.
- `enableDryRun`: (Optional) BETA FEATURE - If present and set to true, enable server-side diff calculations. This feature is in developer preview, and is disabled by default. This config can be specified in the following ways, using this precedence: (1) this `enableDryRun` parameter or (2) the `PULUMI_K8S_ENABLE_DRY_RUN` environment variable.
- `kubeconfig`: (Optional) The contents of a kubeconfig file. If this is set, this config will be used instead of $KUBECONFIG.

> - `namespace`: (Optional) The contents of a kubeconfig file. If this is set, this config will be used instead of $KUBECONFIG.

Reviewer: ```suggestion
- `namespace`: (Optional) The contents of a kubeconfig file. If this is set, this config will be used instead of `$KUBECONFIG`.
```

- `renderYamlToDirectory`: (Optional) BETA FEATURE - If present, render resource manifests to this directory. In this mode, resources will not be created on a Kubernetes cluster, but the rendered manifests will be kept in sync with changes to the Pulumi program. This feature is in developer preview, and is disabled by default. Note that some computed Outputs such as status fields will not be populated since the resources are not created on a Kubernetes cluster. These Output values will remain undefined,
and may result in an error if they are referenced by other resources. Also note that any secret values

> used in these resources will be rendered in plaintext to the resulting YAML.

Reviewer: ```suggestion
used in these resources will be rendered in plain text to the resulting YAML.
```

> - `suppressDeprecationWarnings`: (Optional) If present and set to true, suppress apiVersion deprecation warnings from the CLI. This config can be specified in the following ways, using this precedence: (1) this `suppressDeprecationWarnings` parameter or (2) the `PULUMI_K8S_SUPPRESS_DEPRECATION_WARNINGS` environment variable.

Reviewer: ```suggestion
- `suppressDeprecationWarnings`: (Optional) If present and set to true, suppress `apiVersion` deprecation warnings from the CLI. This config can be specified in the following ways, using this precedence: (1) this `suppressDeprecationWarnings` parameter or (2) the `PULUMI_K8S_SUPPRESS_DEPRECATION_WARNINGS` environment variable.
```
